### PR TITLE
created MyWebAppInitializer, UserController, UserMapper, UserMapperIm…

### PR DESCRIPTION
**Pull request by Alekearl**
…pl, WebConfig

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,24 +17,42 @@
...
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet.jsp</groupId>
+            <artifactId>javax.servlet.jsp-api</artifactId>
+            <version>2.3.3</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (Resci) [inline pom.xml]: Do you use this dependency?

File: pom.xml [hunk 2]
```diff
@@ -49,6 +68,16 @@
             <artifactId>mysql-connector-java</artifactId>
             <version>8.0.17</version>
         </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>jstl</artifactId>
+            <version>1.2</version>
+        </dependency>
```
Comment (Resci) [inline pom.xml]: Same

File: src/main/java/spring/core/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    private UserMapper userMapper;
+    private UserService userService;
+
+    @Autowired
+    public UserController(UserMapper userMapper, UserService userService) {
+        this.userMapper = userMapper;
+        this.userService = userService;
+    }
+
+    @RequestMapping(value = "/user/inject", method = RequestMethod.PUT)
```
Comment (Resci) [inline src/main/java/spring/core/controller/UserController.java]: Why you use put instead get?

File: src/main/java/spring/core/dao/daoimpl/UserDaoImpl.java [hunk 4]
```diff
@@ -40,6 +40,18 @@ public void add(User user) {
         }
     }
 
+    @Override
+    public User get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<User> userQuery =
+                    session.createQuery("FROM User u WHERE u.id = id", User.class);
+            userQuery.setParameter("id", id);
+            return userQuery.getSingleResult();
```
Comment (Resci) [inline src/main/java/spring/core/dao/daoimpl/UserDaoImpl.java]: [Checklist](https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_03.html) p.2

File: src/main/java/spring/core/dto/UserResponseDto.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        this.name = name;
+    }
+
+    @Override
+    public String toString() {
+        return "UserResponseDto{"
+                + "id=" + id
+                + ", name='" + name + '\''
+                + '}';
+    }
```
Comment (Resci) [inline src/main/java/spring/core/dto/UserResponseDto.java:29]: Do you use this method?

File: src/main/java/spring/core/service/serviceimpl/UserMapperImpl.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.springframework.stereotype.Component;
+import spring.core.dto.UserResponseDto;
+import spring.core.model.User;
+import spring.core.service.UserMapper;
+
+@Component
+public class UserMapperImpl implements UserMapper {
+    @Override
+    public UserResponseDto toDto(User user) {
```
Comment (Resci) [inline src/main/java/spring/core/service/serviceimpl/UserMapperImpl.java]:
```suggestion
    public UserResponseDto mapToDto(User user) {
```

File: src/main/java/spring/core/dao/UserDao.java [hunk 7]
```diff
@@ -6,5 +6,7 @@
 public interface UserDao {
     void add(User user);
 
+    User get(Long id);
```
Comment (Resci) [inline src/main/java/spring/core/dao/UserDao.java]:
```suggestion
    Optional<User> get(Long id);
```

**Review by Resci — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (Alekearl) [reply pom.xml]: Emmm. Nope. =)

File: pom.xml (hunk 2 above)
Comment (Alekearl) [reply pom.xml]: Ok. =)

File: src/main/java/spring/core/dto/UserResponseDto.java (hunk 5 above)
Comment (Alekearl) [reply src/main/java/spring/core/dto/UserResponseDto.java:29]: Nope. =)

File: src/main/java/spring/core/controller/UserController.java (hunk 3 above)
Comment (Alekearl) [reply src/main/java/spring/core/controller/UserController.java]: Changed to type request GET. Thanx for explanation. =)

File: src/main/java/spring/core/dao/daoimpl/UserDaoImpl.java (hunk 4 above)
Comment (Alekearl) [reply src/main/java/spring/core/dao/daoimpl/UserDaoImpl.java]: Changed

File: src/main/java/spring/core/controller/UserController.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    private UserMapper userMapper;
+    private UserService userService;
+
+    @Autowired
+    public UserController(UserMapper userMapper, UserService userService) {
+        this.userMapper = userMapper;
+        this.userService = userService;
+    }
+
+    @RequestMapping(value = "/user/inject", method = RequestMethod.GET)
```
Comment (Resci) [inline src/main/java/spring/core/controller/UserController.java]: Checklist p.4

**Review by Resci — CHANGES_REQUESTED** [review]

File: src/main/java/spring/core/controller/UserController.java (hunk 8 above)
Comment (Resci) [reply src/main/java/spring/core/controller/UserController.java:28]: And incorrect url

**Review by Resci — CHANGES_REQUESTED** [review]

**Review by Resci — APPROVED** [review]

File: src/main/java/spring/core/controller/UserController.java (hunk 8 above)
Comment (Alekearl) [reply src/main/java/spring/core/controller/UserController.java]: changed. Thanx for making it more clurifying.

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**